Spec auto-update script worked but not

Context:

- The spec auto-update script worked, since it submitted the following PR:
  - #2769
- _But_ the GH action reported a failure as you can see from the log:
  - https://github.com/open-telemetry/opentelemetry.io/actions/runs/5049440364/jobs/9058896325: for a relevant excerpt see below. The two most relevant lines of that output are:
    - PR _was_ submitted: `Submitting auto-update PR 'Update opentelemetry-specification version to 1.21.0'.`
    - But failed for some reason (access rights, but I'm unsure how/why that would be an issue):
      `pull request update failed: GraphQL: opentelemetrybot does not have permission to update the pull request PR_kwDOCvmNHM5RDIiT. (updatePullRequest)`

_Might_ the failure be because the `git switch` was actually a noop since the spec submodule was _already_ at the specificed tag? Since I'm not sure, I'm creating this issue as a reminder to keep an eye on this around the time that the next spec version will come out.

/cc @svrnm @tigrannajaryan @trask 

```nocode
REPO:            opentelemetry-specification
LATEST VERSION:  1.21.0
SEARCHING for:   '^ *spec:' in scripts/content-modules/adjust-pages.pl
CURRENT VERSION:   spec: 1.20.0
+ sed -i.bak -e 's/\(^ *spec:\) .*/\1 1.21.0/' scripts/content-modules/adjust-pages.pl

Version update necessary:
diff --git a/scripts/content-modules/adjust-pages.pl b/scripts/content-modules/adjust-pages.pl
index e471c40..b944d18 100755
--- a/scripts/content-modules/adjust-pages.pl
+++ b/scripts/content-modules/adjust-pages.pl
@@ -17,7 +17,7 @@ my $semConvRef = "$otelSpecRepoUrl/blob/main/semantic_conventions/README.md";
 my $specBasePath = '/docs/specs';
 my $path_base_for_github_subdir = "content/en$specBasePath";
 my %versions = qw(
-  spec: 1.20.0
+  spec: 1.21.0
   otlp: main
 );
 my $otelSpecVers = $versions{'spec:'};

Switching to opentelemetry-specification at tag v1.21.0
+ npm run get:submodule -- content-modules/opentelemetry-specification

> get:submodule
> npm run _get:${GET:-submodule} content-modules/opentelemetry-specification


> _get:submodule
> set -x && git submodule update --init ${DEPTH:- --depth 1} content-modules/opentelemetry-specification

+ git submodule update --init --depth 1 content-modules/opentelemetry-specification
Submodule 'content-modules/opentelemetry-specification' (https://github.com/open-telemetry/opentelemetry-specification.git) registered for path 'content-modules/opentelemetry-specification'
Cloning into '/home/runner/work/opentelemetry.io/opentelemetry.io/content-modules/opentelemetry-specification'...
From https://github.com/open-telemetry/opentelemetry-specification
 * branch            4229197f45f0da0642bc7c0127e84c8aa1439489 -> FETCH_HEAD
Submodule path 'content-modules/opentelemetry-specification': checked out '4229197f45f0da0642bc7c0127e84c8aa1439489'
+ cd content-modules/opentelemetry-specification
+ git fetch
From https://github.com/open-telemetry/opentelemetry-specification
 * [new tag]         v1.21.0    -> v1.21.0
+ git switch --detach v1.21.0
HEAD is now at 4229197 Release 1.21.0 (#[34](https://github.com/open-telemetry/opentelemetry.io/actions/runs/5049440364/jobs/9058896325#step:4:35)74)
Switched to a new branch 'opentelemetrybot/auto-update-opentelemetry-specification-1.21.0'
[opentelemetrybot/auto-update-opentelemetry-specification-1.21.0 db[35](https://github.com/open-telemetry/opentelemetry.io/actions/runs/5049440364/jobs/9058896325#step:4:36)3[47](https://github.com/open-telemetry/opentelemetry.io/actions/runs/5049440364/jobs/9058896325#step:4:48)] Update opentelemetry-specification version to 1.21.0
 1 file changed, 1 insertion(+), 1 deletion(-)
remote: Resolving deltas:   0% (0/2)        
remote: Resolving deltas:  [50](https://github.com/open-telemetry/opentelemetry.io/actions/runs/5049440364/jobs/9058896325#step:4:51)% (1/2)        
remote: Resolving deltas: 100% (2/2)        
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.        
remote: 
remote: Create a pull request for 'opentelemetrybot/auto-update-opentelemetry-specification-1.21.0' on GitHub by visiting:        
remote:      https://github.com/open-telemetry/opentelemetry.io/pull/new/opentelemetrybot/auto-update-opentelemetry-specification-1.21.0        
remote: 
To https://github.com/open-telemetry/opentelemetry.io
 * [new branch]      opentelemetrybot/auto-update-opentelemetry-specification-1.21.0 -> opentelemetrybot/auto-update-opentelemetry-specification-1.21.0
branch 'opentelemetrybot/auto-update-opentelemetry-specification-1.21.0' set up to track 'origin/opentelemetrybot/auto-update-opentelemetry-specification-1.21.0'.
Submitting auto-update PR 'Update opentelemetry-specification version to 1.21.0'.
pull request update failed: GraphQL: opentelemetrybot does not have permission to update the pull request PR_kwDOCvmNHM5RDIiT. (updatePullRequest)
https://github.com/open-telemetry/opentelemetry.io/pull/27[69](https://github.com/open-telemetry/opentelemetry.io/actions/runs/5049440364/jobs/9058896325#step:4:70)
Error: Process completed with exit code 1.
```

--- a/.github/workflows/scripts/auto-update-version.sh
+++ b/.github/workflows/scripts/auto-update-version.sh
@@ -86,6 +86,4 @@ $GIT commit -a -m "$message"
 $GIT push --set-upstream origin "$branch"
 
 echo "Submitting auto-update PR '$message'."
-$GH pr create --label auto-update \
-             --title "$message" \
-             --body "$body"
+$GH pr create --title "$message" --body "$body"

